hastory-cli/src/Hastory/Cli/OptParse/Types.hs

```diff
@@ -61,6 +61,8 @@ data Settings =
     }
   deriving (Show, Eq)
 
+type RemoteStorageUrl = T.Text
```

Comment:
You can just use `BaseUrl` here, I think.

Reply:
Oh, thanks. Indeed, parsing this early-on is the best practice and eliminates much of the ceremony afterwards. Modified the code now.